utils/pre-publish-check-ci.ts

```diff
@@ -16,16 +16,25 @@ import {
 // eslint-disable-next-line promise/catch-or-return
 fg(path.join(__dirname, "..", "packages", "*", "package.json")).then(
     (pkgPaths) => {
+        let allPassed = true;
         // eslint-disable-next-line promise/always-return
         for (const pkgPath of pkgPaths) {
             // eslint-disable-next-line @typescript-eslint/no-require-imports
             const pkgJson = require(path.relative(__dirname, pkgPath));
 
-            if (!checkPrivate(pkgJson)) {
-                checkPublishConfig(pkgJson);
-                checkEntrypoints(pkgJson);
-                checkSource(pkgJson);
+            if (
+                !checkPrivate(pkgJson) &&
+                !checkPublishConfig(pkgJson) &&
+                !checkEntrypoints(pkgJson) &&
+                !checkSource(pkgJson)
+            ) {
+                allPassed = false;
             }
         }
+
+        // Exit only after we've processed all the packages.
+        if (!allPassed) {
+            process.exit(1);
+        }
```

Comment:
**note:** This means we check all the packages for all the things before we quit, rather than quitting on the first error. Makes it easier to do changes to the overall checks and ensure compliance.